core/src/com/unciv/logic/multiplayer/apiv2/ApiV2Wrapper.kt

```diff
@@ -0,0 +1,255 @@
+
+    /**
+     * API for lobby management
+     */
+    val lobby = LobbyApi(client, authHelper)
+
+    /**
+     * Start a new WebSocket connection
+     *
+     * The parameter [handler] is a coroutine that will be fed the established
+     * [ClientWebSocketSession] on success at a later point. Note that this
+     * method does instantly return, detaching the creation of the WebSocket.
+     * The [handler] coroutine might not get called, if opening the WS fails.
+     * Use [jobCallback] to receive the newly created job handling the WS connection.
+     */
+    internal suspend fun websocket(handler: suspend (ClientWebSocketSession) -> Unit, jobCallback: ((Job) -> Unit)? = null): Boolean {
+        Log.debug("Starting a new WebSocket connection ...")
+
+        coroutineScope {
+            try {
+                val session = client.webSocketRawSession {
+                    authHelper.add(this)
+                    url {
+                        protocol = if (Url(baseServer).protocol.isSecure()) URLProtocol.WSS else URLProtocol.WS
+                        port = Url(baseServer).specifiedPort.takeUnless { it == DEFAULT_PORT } ?: protocol.defaultPort
+                        appendPathSegments("api/v2/ws")
+                    }
+                }
+                val job = Concurrency.runOnNonDaemonThreadPool {
+                    handler(session)
+                }
+                websocketJobs.add(job)
+                Log.debug("A new WebSocket has been created, running in job $job")
+                if (jobCallback != null) {
+                    jobCallback(job)
+                }
+                true
+            } catch (e: SerializationException) {
+                Log.debug("Failed to create a WebSocket: %s", e.localizedMessage)
+                return@coroutineScope false
+            } catch (e: Exception) {
+                Log.debug("Failed to establish WebSocket connection: %s", e.localizedMessage)
+                return@coroutineScope false
+            }
+        }
+
+        return true
+    }
+
+    /**
+     * Retrieve the currently available API version of the connected server
+     *
+     * Unlike other API endpoint implementations, this function does not handle
+     * any errors or retries on failure. You must wrap any call in a try-except
+     * clause expecting any type of error. The error may not be appropriate to
+     * be shown to end users, i.e. it's definitively no [UncivShowableException].
+     */
+    suspend fun version(): VersionResponse {
+        return client.get("api/version").body()
+    }
+
+}
+
+/**
+ * APIv2 exception class that is compatible with [UncivShowableException]
+ */
+class ApiException(val error: ApiErrorResponse) : UncivShowableException(lookupErrorMessage(error.statusCode))
+
+/**
+ * Convert an API status code to a string that can be translated and shown to users
+ */
+private fun lookupErrorMessage(statusCode: ApiStatusCode): String {
```

Comment:
Since every status code needs a user-readable string, why not just put these values in the `enum class ApiStatusCode`?

Reply:
(This is not a merge blocker I intent to merge at the next version release)

Reply:
You mean by `override fun toString(): String`? Yes, that would be much cleaner.